src/status_im/transport/inbox.cljs

```diff
@@ -130,119 +92,138 @@
+   changed to error if it is not connected by then"
+  [{:keys [db] :as cofx}]
+  (let [web3     (:web3 db)
+        wnode    (get-current-wnode-address db)]
+    (when config/offline-inbox-enabled?
+      (handlers-macro/merge-fx cofx
+                               {::add-peer {:wnode wnode}
+                                :utils/dispatch-later [{:ms connection-timeout :dispatch [:inbox/connection-check]}]}
+                               (update-mailserver-status :connecting)
+                               (generate-mailserver-symkey)))))
+
+(defn peers-summary-change-fx
+  "Called when a peer summary signal is received or we want to try to connect to mailserver ie. after login.
+   Marks the mailserver as trusted peer if it is already added to the list of peers or connects to it if it's
+   not the case"
+  [{:keys [db] :as cofx}]
+  (if (and (:account/account db)
+           config/offline-inbox-enabled?)
+    (let [{:keys [peers-summary peers-count]} db]
+      (if (zero? peers-count)
+        (update-mailserver-status :disconnected cofx)
+        (let [wnode (get-current-wnode-address db)
+              mailserver-status (:mailserver-status db)]
+          (if (registered-peer? peers-summary wnode)
+            (handlers-macro/merge-fx cofx
+                                     {::mark-trusted-peer {:web3  (:web3 db)
+                                                           :wnode wnode}}
+                                     (generate-mailserver-symkey))
+            (connect-to-mailserver cofx)))))))
+
+(defn get-topics
+  [db topics discover?]
+  (let [inbox-topics    (:inbox/topics db)
+        discovery-topic (transport.utils/get-topic constants/contact-discovery)
+        topics          (or topics
+                            (map #(:topic %) (vals (:transport/chats db))))]
+    (cond-> (apply conj inbox-topics topics)
+      discover? (conj discovery-topic))))
+
+(defn request-messages [{:keys [topics discover? should-recover?]
+                         :or {should-recover? true
+                              discover?       true}}
+                        {:keys [db] :as cofx}]
+  (let [mailserver-status (:mailserver-status db)
+        sym-key-id        (:inbox/sym-key-id db)
+        wnode             (get-current-wnode-address db)
+        inbox-topics      (get-topics db topics discover?)
+        inbox-ready?      (and (= :connected mailserver-status)
+                               sym-key-id)]
+    (when should-recover?
+      (if inbox-ready?
+        {::request-messages {:wnode      wnode
+                             :topics     (into [] inbox-topics)
+                             :sym-key-id sym-key-id
+                             :web3       (:web3 db)}
+         :db                (assoc db :inbox/fetching? true)
+         :dispatch-later    [{:ms fetching-messages-notification-timeout
+                              :dispatch [:inbox/remove-fetching-notification]}]}
```

Comment:
Maybe also dispatch `:inbox/remove-fetching-notification` on `.requestMessages` success event as well? To show user that we finished with fetching.